base_layer/core/src/proof_of_work/randomx_factory.rs

```
@@ -118,12 +131,40 @@ impl RandomXFactoryInner {
             }
         }
 
-        // TODO: put config in.
-
-        let vm = RandomXVMInstance::new(&key)?;
+        let vm = RandomXVMInstance::create(&key, self.config.into())?;
 
         self.vms.insert(Vec::from(key), (Instant::now(), vm.clone()));
 
         Ok(vm)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn basic_initialization_and_hash() {
+        let factory = RandomXFactory::new(Default::default(), 2);
+
+        let key = b"some-key";
+        let vm = factory.create(&key[..]).unwrap();
+        let preimage = b"hashme";
+        let hash1 = vm.calculate_hash(&preimage[..]).unwrap();
+        let vm = factory.create(&key[..]).unwrap();
+        assert_eq!(vm.calculate_hash(&preimage[..]).unwrap(), hash1);
+
+        let key = b"another-key";
+        let vm = factory.create(&key[..]).unwrap();
+        assert_ne!(vm.calculate_hash(&preimage[..]).unwrap(), hash1);
+    }
+
+    #[test]
+    fn large_page_fallback() {
+        // This only tests the fallback branch on platforms that do not support large pages (e.g. MacOS)
+        let factory = RandomXFactory::new(RandomXConfig { use_large_pages: true }, 1);
```

We should remove `use_large_pages`, it's an easy error to make to set this to true where it should be false.

RandomXFlag::get_recommended_flags() is the recommended way to use the library (perhaps fall back to RandomXFlag::FLAG_DEFAULT if this fails for any reason).